experimental/desktop_photo_search/lib/main.dart

```diff
@@ -101,20 +102,21 @@ class UnsplashHomePage extends StatelessWidget {
                     ? PhotoDetails(
                         photo: photoSearchModel.selectedPhoto,
                         onPhotoSave: (photo) async {
-                          final result = await file_chooser.showSavePanel(
-                            suggestedFileName: '${photo.id}.jpg',
-                            allowedFileTypes: const [
-                              file_chooser.FileTypeFilterGroup(
-                                label: 'JPGs',
-                                fileExtensions: ['jpg'],
-                              )
+                          final path = await getSavePath(
+                            suggestedName: '${photo.id}.jpg',
+                            acceptedTypeGroups: <XTypeGroup>[
+                              XTypeGroup(
+                                label: 'JPG',
+                                extensions: ['jpg'],
+                                mimeTypes: ['image/jpeg'],
+                              ),
                             ],
                           );
-                          if (!result.canceled) {
-                            final bytes =
-                                await photoSearchModel.download(photo: photo);
-                            await File(result.paths[0]).writeAsBytes(bytes);
-                          }
+                          final fileData =
+                              await photoSearchModel.download(photo: photo);
+                          final photoFile =
+                              XFile.fromData(fileData, mimeType: 'image/jpeg');
+                          photoFile.saveTo(path);
```

Comment:
What happens if the user cancels the save dialog? You've removed the logic that handled that.

Reply:
It wasn't obvious from the types that the returned string was nullable, and that is how file save cancellation is now being communicated. This will be better when null safety is a thing.

Reply:
> It wasn't obvious from the types that the returned string was nullable, and that is how file save cancellation is now being communicated.

Yes, file_chooser had an explicit cancel status since I thought the added clarity was worth the slightly more complex return type, but 🤷